From: Daniel Leidert <dleidert@debian.org>
Date: Mon, 3 Feb 2020 12:27:52 +0100
Subject: Fix compatibility with i18n 1.x

Allow ruby-i18n >= 0.7 but < 2 and fix some warnings about instance variables
not initialized.

---
 lib/stringex/localization/backend/i18n.rb     | 2 +-
 lib/stringex/localization/backend/internal.rb | 6 +++---
 stringex.gemspec                              | 6 +++---
 3 files changed, 7 insertions(+), 7 deletions(-)

--- a/lib/stringex/localization/backend/i18n.rb
+++ b/lib/stringex/localization/backend/i18n.rb
@@ -12,7 +12,7 @@
           end
 
           def locale
-            @locale || ::I18n.locale
+            defined?(@locale) && @locale ? @locale : ::I18n.locale
           end
 
           def locale=(new_locale)
--- a/lib/stringex/localization/backend/internal.rb
+++ b/lib/stringex/localization/backend/internal.rb
@@ -6,7 +6,7 @@
 
         class << self
           def locale
-            @locale || default_locale
+            defined?(@locale) && @locale ? @locale : default_locale
           end
 
           def locale=(new_locale)
@@ -14,7 +14,7 @@
           end
 
           def default_locale
-            @default_locale || DEFAULT_LOCALE
+            defined?(@default_locale) && @default_locale ? @default_locale : DEFAULT_LOCALE
           end
 
           def default_locale=(new_locale)
@@ -44,4 +44,4 @@
       end
     end
   end
-end
\ No newline at end of file
+end
--- a/stringex.gemspec
+++ b/stringex.gemspec
@@ -287,7 +287,7 @@
       s.add_development_dependency(%q<sqlite3>.freeze, ["= 1.3.10"])
       s.add_development_dependency(%q<test-unit>.freeze, ["= 3.0.9"])
       s.add_development_dependency(%q<activerecord>.freeze, ["= 5.1.4"])
-      s.add_development_dependency(%q<i18n>.freeze, ["= 0.7.0"])
+      s.add_development_dependency(%q<i18n>.freeze, [">= 0.7", "< 2"])
     else
       s.add_dependency(%q<jeweler>.freeze, ["= 2.3.7"])
       s.add_dependency(%q<travis-lint>.freeze, ["= 1.7.0"])
@@ -295,7 +295,7 @@
       s.add_dependency(%q<sqlite3>.freeze, ["= 1.3.10"])
       s.add_dependency(%q<test-unit>.freeze, ["= 3.0.9"])
       s.add_dependency(%q<activerecord>.freeze, ["= 5.1.4"])
-      s.add_dependency(%q<i18n>.freeze, ["= 0.7.0"])
+      s.add_dependency(%q<i18n>.freeze, [">= 0.7", "< 2"])
     end
   else
     s.add_dependency(%q<jeweler>.freeze, ["= 2.3.7"])
@@ -304,7 +304,7 @@
     s.add_dependency(%q<sqlite3>.freeze, ["= 1.3.10"])
     s.add_dependency(%q<test-unit>.freeze, ["= 3.0.9"])
     s.add_dependency(%q<activerecord>.freeze, ["= 5.1.4"])
-    s.add_dependency(%q<i18n>.freeze, ["= 0.7.0"])
+    s.add_dependency(%q<i18n>.freeze, [">= 0.7", "< 2"])
   end
 end
 
